middle_end/closure/closure.ml

```diff
@@ -798,15 +798,16 @@ let direct_apply env fundesc ufunct uargs ~loc ~attribute =
   then app
   else Usequence(ufunct, app)
 
-(* Add [Value_integer] info to the approximation of an application *)
+(* Add [Value_integer] or [Value_constptr] info to the approximation
+   of an application *)
 
 let strengthen_approx appl approx =
   match approx_ulam appl with
     (Value_const _) as intapprox ->
       intapprox
   | _ -> approx
 
-(* If a term has approximation Value_integer and is pure,
+(* If a term has approximation Value_integer or Value_constptr and is pure,
    replace it by an integer constant *)
```

Comment:
I think this is the wrong resolution (likewise for lines 801/802 above).  We should be removing the "or Value_constptr" phrase.